roles/openshift_cfme/tasks/validate.yml

```diff
@@ -0,0 +1,74 @@
+---
+# Validate configuration parameters passed to the openshift_cfme role
+
+######################################################################
+# CORE PARAMETERS
+- name: Ensure openshift_cfme_app_template is valid
+  assert:
+    that:
+      - openshift_cfme_app_template in __openshift_cfme_app_templates
+    msg: "openshift_cfme_app_template must be one of {{ __openshift_cfme_app_templates | join(', ') }}"
+
+- name: Ensure openshift_cfme_storage_class is a valid type
+  assert:
+    that:
+      - openshift_cfme_storage_class in __openshift_cfme_storage_classes
+    msg: "openshift_cfme_storage_class must be one of {{ __openshift_cfme_storage_classes | join(', ') }}"
+
+######################################################################
+# STORAGE PARAMS - NFS
+- name: Ensure external NFS storage has a valid NFS server hostname defined
+  assert:
+    that:
+      - openshift_cfme_storage_nfs_external_hostname | default(False)
+    msg: The selected storage class 'nfs_external' requires a valid hostname for the openshift_cfme_storage_nfs_hostname parameter
+  when:
+    - openshift_cfme_storage_class == 'nfs_external'
+
+######################################################################
+# STORAGE PARAMS  -CLOUD PROVIDER
+
+- name: Validate Cloud Provider storage class
+  assert:
+    that:
+      - openshift_cloudprovider_kind == 'aws' or openshift_cloudprovider_kind == 'gce'
+    msg: |
+      openshift_cfme_storage_class is 'cloudprovider' but you have an
+      invalid kind defined. See 'openshift_cloudprovider_kind' in the
+      example inventories for the required parameters for your
+      selected cloud provider. Working providers: 'aws' and 'gce'.
+  when:
+    - openshift_cfme_storage_class == 'cloudprovider'
+    - openshift_cloudprovider_kind is defined
+
+- name: Validate 'cloudprovider' Storage Class has required parameters defined
+  assert:
+    that:
+      - not openshift_cloudprovider_kind is defined
```

Comment:
This is blowing up on me when i have openshift_cfme_storage_class set to 'cloudprovider'. My inventory does have openshift_cloudprovider_kind set to 'aws'. It looks like the logic is simply wrong. Should be
```
that:
- openshift_cloudprovider_kind is defined
```